migrations/lib/converters/discourse/steps/groups.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Migrations::Converters::Discourse
+  class Groups < ::Migrations::Converters::Base::ProgressStep
+    attr_accessor :source_db
+
+    def max_progress
+      @source_db.count <<~SQL
+        SELECT COUNT(*) FROM groups
+      SQL
+    end
+
+    def items
+      @source_db.query <<~SQL
+        SELECT g.*,
+               u.id AS original_flair_upload_id,
+               u.url AS flair_path,
+               u.original_filename AS flair_filename,
+               CASE WHEN g.automatic THEN g.name ELSE NULL END AS existing_id
```

Comment:
`ELSE NULL` at the end of a `CASE` is redundant

```suggestion
               CASE WHEN g.automatic THEN g.name END AS existing_id
```